### PR TITLE
2111 cloud native liberty buildpack

### PR DESCRIPTION
**Pull request by kevin-ortega**
For #2111 
Looking to publish on 04/01/2022

### CODE REVIEW

**Comment by kevin-ortega** [comment]
Updated image: 03/28/2022

image

**Comment by kevin-ortega** [comment]
https://draft-openlibertyio.mybluemix.net/2022/03/25/2022-03-31-cloud-native-liberty-buildpack.html

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 1]
```diff
@@ -0,0 +1,124 @@
+---
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc:1]: there is some kind of problem with this metadata section but I can't tell exactly what it is. If you look at the GitHub file view, the section does not format to be a table as expected. Also, your github pic and link aren't working, nor is hankl showing up as an author, despite all the URLs being correct

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+layout: post
+title: Introducing the Paketo Cloud Native Liberty Buildpack
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/kevin-ortega
+author_github: https://github.com/kevin-ortega
+seo-title: Introducing a new method to deploy your cloud native application to any cloud - paketo.io
+seo-description: Use Paketo Buildpacks to easily build your apps and keep them updated.  Buildpacks transform your application source code into container images without creating a Dockerfile!
+blog_description: Use Paketo Buildpacks to easily build your apps and keep them updated.  Buildpacks transform your application
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: blog description value needs to be enclosed in quotes

Comment (dmuelle) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: also, only need one space after "keep them updated."

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 3]
```diff
@@ -0,0 +1,124 @@
+---
+layout: post
+title: Introducing the Paketo Cloud Native Liberty Buildpack
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/kevin-ortega
+author_github: https://github.com/kevin-ortega
+seo-title: Introducing a new method to deploy your cloud native application to any cloud - paketo.io
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: normally this field ends with "openliberty.io" Im not sure if that's part of the problem

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around
+// screenshots but not around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Once again make sure to double check how it looks
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+link:https://paketo.io/[Paketo buildpacks] transform your application source code into container images and easily keeps them updated.
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: suggested edit:
link:https://paketo.io/[Paketo buildpacks] transform your application source code into container images and easily keep those images updated.

this is correct but a little awkward (buildpacks...keep)

you could also rephrase  to something like

link:https://paketo.io/[Paketo buildpacks] transform your application source code into easily maintained container images.

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+link:https://paketo.io/[Paketo buildpacks] transform your application source code into container images and easily keeps them updated.  
+
+== What is a buildpack?
+
+A buildpack is a set of executables that inspects your app source code and creates a plan to build and run your application.  
+
+Paketo buildpacks are toolkits and workflows for building reproducible container images from source code.
+
+The Paketo Cloud Native Liberty buildpack provides the Open Liberty runtime to a workflow that produces an OCI image that can run just about anywhere.
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
The link:https://github.com/paketo-buildpacks/liberty[Paketo Cloud Native Liberty buildpack] provides the Open Liberty runtime to a workflow that produces an Open Container Initiative (OCI) image that can run just about anywhere.
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Once again make sure to double check how it looks
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+link:https://paketo.io/[Paketo buildpacks] transform your application source code into container images and easily keeps them updated.  
+
+== What is a buildpack?
+
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc:62]: recommend removing the spaces between lines

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+== What do you need?
+* Your application source
+
+* link:https://hub.docker.com/search?type=edition&offering=community[Docker]
+
+* The link:https://buildpacks.io/docs/tools/pack/[pack CLI]
+
+== Let's get started!
+
+Download the sample getting started app:
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
Download the Open Liberty starter app:
```

Comment (dmuelle) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: Is this example using the OL starter app as "your application source". If so, there should be some kind of explanation- the prereq list suggests they should supply their own source. Or does the starter integrate with the user source app?

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+Build the app:
+[source]
+pack build myapp --env BP_JAVA_APP_SERVER=liberty
+
+Your application now is ready to run. 
+
+[source]
+
+== You can build from an on-prem Open Liberty installation
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
== Build your app from an on-prem Open Liberty installation
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+Using a Liberty packaged server
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
You can build from an on-prem Open Liberty installation by using a packaged Liberty server. Run the following command to package your server.
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+[source]
+bin/server package defaultServer --include=usr
+
+The packaged server can then be supplied to the build using the `--path` argument:
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
You can then supply the packaged server to the build by using the `--path` argument:
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+[source]
+pack build --path <packaged-server-zip-path> --env BP_JAVA_APP_SERVER=liberty myapp 
+
+Or to build from a Liberty server installation, change your working directory to the installation root containing the wlp directory and run:
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
Alternatively, you can build from a Liberty server installation by changing your working directory to the installation root that contains the `wlp` directory and running the following command:
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+
+Set a default builder:
+[source]
+pack config default-builder paketobuildpacks/builder:base
+
+Build the app:
+[source]
+pack build myapp --env BP_JAVA_APP_SERVER=liberty
+
+Your application now is ready to run.
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: is it now in an OCI image that can run anywhere? If so, we should say, if not, what other steps are needed?

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- made a few suggestions, let me know if you have any questions. Thanks!

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 2 above)
Comment (kevin-ortega) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: I made these edits.

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 4 above)
Comment (kevin-ortega) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: Thanks, I've made this edit.

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 3 above)
Comment (kevin-ortega) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: I've made this edit.

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 6 above)
Comment (kevin-ortega) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc:62]: done

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 12 above)
Comment (kevin-ortega) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]: Yes, it's now an OCI image.   I reworded that sentence.

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 4 above)
Comment (dmuelle) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
link:https://paketo.io/[Paketo buildpacks] transform your application source code into container images and easily keep them updated.  
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
Your application is now transformed into an OCI image that is ready to run almost anywhere. 
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 14]
```diff
@@ -0,0 +1,124 @@
...
+
+== What do you need?
+* Your application source
+
+* link:https://hub.docker.com/search?type=edition&offering=community[Docker]
+
+* The link:https://buildpacks.io/docs/tools/pack/[pack CLI]
+
+== Let's get started!
+
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
The following examples use the link:/blog/2021/08/20/open-liberty-starter.html[Open Liberty starter] app as the application source.
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 15]
```diff
@@ -0,0 +1,124 @@
+---
+layout: post
+title: "Introducing the Paketo Cloud Native Liberty Buildpack"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/kevin-ortega
+author_github: https://github.com/kevin-ortega
+seo-title: Introducing a new method to deploy your cloud native application to any cloud - OpenLiberty.io
+seo-description: Use Paketo Buildpacks to easily build your apps and keep them updated. Buildpacks transform your application source code into container images without creating a Dockerfile!
+blog_description: "Use Paketo Buildpacks to easily build your apps and keep them updated. Buildpacks transform your application"
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
blog_description: "Use Paketo Buildpacks to easily build your apps and keep them updated. Buildpacks transform your application source code into container images without creating a Dockerfile!" 
```

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc [hunk 16]
```diff
@@ -0,0 +1,124 @@
...
+layout: post
+title: "Introducing the Paketo Cloud Native Liberty Buildpack"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/kevin-ortega
+author_github: https://github.com/kevin-ortega
+seo-title: Introducing a new method to deploy your cloud native application to any cloud - OpenLiberty.io
+seo-description: Use Paketo Buildpacks to easily build your apps and keep them updated. Buildpacks transform your application source code into container images without creating a Dockerfile!
+blog_description: "Use Paketo Buildpacks to easily build your apps and keep them updated. Buildpacks transform your application" 
+source code into container images without creating a Dockerfile!
```
Comment (dmuelle) [inline posts/2022-04-01-cloud-native-liberty-buildpack.adoc]:
```suggestion
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
fixing the metadata and a few more minor things

File: posts/2022-04-01-cloud-native-liberty-buildpack.adoc (hunk 1 above)
Comment (kevin-ortega) [reply posts/2022-04-01-cloud-native-liberty-buildpack.adoc:1]: I think this is resolved.

**Comment by kevin-ortega** [comment]
@dmuelle Thank you for your help with the metadata problem and all your great suggested edits.

**Review by dmuelle — DISMISSED** [review]

**Review by dmuelle — DISMISSED** [review]

**Review by dmuelle — APPROVED** [review]
LGTM- just need to commit this change back to draft as well

**Comment by kevin-ortega** [comment]
> 

This was actually changed in draft first...so it's already there.

**Pull request merged**